### ISSUE TITLE
Example smoke test failures under PyTorch 2.0

### ISSUE THREAD

**Issue by eb8680** [pyro-ppl/pyro#3215]
From #3192, the AIR and baseball examples crash under `torch>=2.0.0`:
- [x] `tests/test_examples.py::test_cpu[air/main.py --num-steps=1]`
- [x] `tests/test_examples.py::test_cpu[air/main.py --num-steps=1 --no-baseline]`
- [x] `tests/test_examples.py::test_jit[air/main.py --num-steps=1 --jit]`
- [x] `tests/test_examples.py::test_cpu[baseball.py --num-samples=200 --warmup-steps=100 --num-chains=2]`

 It seems like the failures are related to minor API changes in data loading utilities and should be easy to fix:

Traceback from the first AIR test case:
<details>

```
Traceback (most recent call last):
  File "/home/runner/work/pyro/pyro/examples/air/main.py", line 453, in <module>
    main(**vars(parser.parse_args()))
  File "/home/runner/work/pyro/pyro/examples/air/main.py", line 140, in main
    X, true_counts = load_data()
  File "/home/runner/work/pyro/pyro/examples/air/main.py", line 120, in load_data
    X_np, Y = multi_mnist.load(inpath)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyro/contrib/examples/multi_mnist.py", line 92, in load
    np.savez_compressed(f, x=x, y=y)
  File "<__array_function__ internals>", line 200, in savez_compressed
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/numpy/lib/npyio.py", line 686, in savez_compressed
    _savez(file, args, kwds, True)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/numpy/lib/npyio.py", line 716, in _savez
    val = np.asanyarray(val)
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (60000,) + inhomogeneous part.
Exception ignored in: <function ZipFile.__del__ at 0x7f484[58](https://github.com/pyro-ppl/pyro/actions/runs/5008810154/jobs/8977153761?pr=3192#step:6:59)83940>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.16/x[64](https://github.com/pyro-ppl/pyro/actions/runs/5008810154/jobs/8977153761?pr=3192#step:6:65)/lib/python3.8/zipfile.py", line 1821, in __del__
    self.close()
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/zipfile.py", line 1838, in close
    self.fp.seek(self.start_dir)
ValueError: seek of closed file
```
</details>

And in subsequent AIR test cases:
<details>

```
Traceback (most recent call last):
  File "/home/runner/work/pyro/pyro/examples/air/main.py", line 453, in <module>
    main(**vars(parser.parse_args()))
  File "/home/runner/work/pyro/pyro/examples/air/main.py", line 140, in main
    X, true_counts = load_data()
  File "/home/runner/work/pyro/pyro/examples/air/main.py", line 120, in load_data
    X_np, Y = multi_mnist.load(inpath)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pyro/contrib/examples/multi_mnist.py", line 78, in load
    data = np.load(file_path, allow_pickle=True)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/numpy/lib/npyio.py", line 420, in load
    ret = NpzFile(fid, own_fid=own_fid, allow_pickle=allow_pickle,
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/numpy/lib/npyio.py", line 187, in __init__
    _zip = zipfile_factory(fid)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/numpy/lib/npyio.py", line 103, in zipfile_factory
    return zipfile.ZipFile(file, *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/zipfile.py", line 1269, in __init__
    self._RealGetContents()
  File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/zipfile.py", line 1336, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```
</details>

And the baseball example:
<details>

```
Traceback (most recent call last):
  File "/home/runner/work/pyro/pyro/examples/baseball.py", line 423, in <module>
    main(args)
  File "/home/runner/work/pyro/pyro/examples/baseball.py", line 256, in main
    baseball_dataset = pd.read_csv(DATA_URL, "\t")
TypeError: read_csv() takes 1 positional argument but 2 were given
```
</details>

**Comment by martinjankowiak** [comment]
baseball was an easy fix but i'm not sure what's going on with multi_mnist...

**Comment by martinjankowiak** [comment]
ok `test_examples.py` tests have been fixed; results have been merged into #3192.

afaict the needed air change was an numpy issue (?)

on the other hand seeing this, not sure if this is just flaky

```tutorial/source/prodlda::ipynb::Cell 6 FAILED                            [ 76%]

=================================== FAILURES ===================================
____________________ tutorial/source/prodlda.ipynb::Cell 6 _____________________
Notebook cell execution failed
Cell 6: Cell execution caused an exception

Input:
news = fetch_20newsgroups(subset='all')
vectorizer = CountVectorizer(max_df=0.5, min_df=20, stop_words='english')
docs = torch.from_numpy(vectorizer.fit_transform(news['data']).toarray())

vocab = pd.DataFrame(columns=['word', 'index'])
vocab['word'] = vectorizer.get_feature_names()
vocab['index'] = vocab.index
```

**Comment by martinjankowiak** [comment]
closing since the examples flow now passes
